code/game/objects/items/weapons/mop.dm

```diff
@@ -78,3 +78,33 @@
 	throwforce = 8
 	throw_range = 4
 	mopspeed = 20
+	var/refill_enabled = TRUE //Self-refill toggle for when a janitor decides to mop with something other than water.
+	var/refill_rate = 1 //Rate per process() tick mop refills itself
+	var/refill_reagent = "water" //Determins what reagent to use for refilling, just in case someone wanted to make a HOLY MOP OF PURGING
+
+/obj/item/weapon/mop/advanced/New()
+	..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/weapon/mop/advanced/attack_self(mob/user)
+	refill_enabled = !refill_enabled
+	if(refill_enabled)
+		START_PROCESSING(SSobj, src)
+	else
+		STOP_PROCESSING(SSobj,src)
+	user << "<span class='notice'>You turn [refill_enabled ? "on" : "off"] the condenser.</span>"
```

Comment:
Wouldn't it look better being: "You turn the condenser [on : off]." ? Maybe it's just me.